### PR TITLE
hellmaker: init 0-unstable-2023-03-18

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by superherointj** [comment]
* Please follow commit conventions:

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

* Title of PR needs fixing too.

* Check version format.

I won't be reviewing this, but I'm advancing problems for you.

File: pkgs/by-name/he/hellmaker/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchFromGitHub {
+    owner = "0xNinjaCyclone";
+    repo = "hellMaker";
+    rev = "2e9efe2aed9412f67a8606544392a2df10d3a2d0";
+    hash = "sha256-zbtzlYGCLW/lt7GJvMHao/MZhdghNBQCQsjUImL1RC4=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
```
Comment (superherointj) [inline pkgs/by-name/he/hellmaker/package.nix:21]: Install hooks are missing.

File: pkgs/by-name/he/hellmaker/package.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    mkdir -p $out/{bin,share/hellmaker}
+    cp -R * $out/share/hellmaker
+    makeWrapper ${python3.interpreter} $out/bin/hellmaker \
+      --set PYTHONPATH "$PYTHONPATH:$out/share/hellmaker/hellMaker.py" \
+      --add-flags "$out/share/hellmaker/hellMaker.py"
+  '';
+
+  meta = with lib; {
```
Comment (superherointj) [inline pkgs/by-name/he/hellmaker/package.nix:29]:
```suggestion
  meta = {
```
Avoid nested `with`.

File: pkgs/by-name/he/hellmaker/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hell-maker";
+  version = "unstable-2023-03-18";
```
Comment (superherointj) [inline pkgs/by-name/he/hellmaker/package.nix:10]:
```suggestion
  version = "unstable-2023-03-18";
```
Recently unstable version also includes last stable tag.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

File: pkgs/by-name/he/hellmaker/package.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hell-maker";
```
Comment (superherointj) [inline pkgs/by-name/he/hellmaker/package.nix:9]:
```suggestion
  pname = "hellmaker";
```
This needs to be equal to top-level name. In this case `hellmaker`.

**Review by nicoonoclaste — DISMISSED** [review]

File: pkgs/by-name/he/hellmaker/package.nix (hunk 2 above)
Comment (superherointj) [reply pkgs/by-name/he/hellmaker/package.nix:31]: Are you going to do this change?

Comment (Tochiaha) [reply pkgs/by-name/he/hellmaker/package.nix]: changes have been applied.

**Review by superherointj — APPROVED** [review]
Package wise looks good.

**Pull request merged**